### DISCUSSION TITLE
Pull Request was force-pushed even though the PR was edited and a rebase was not requested

### DISCUSSION THREAD

**Discussion by Keysox** [renovatebot/renovate#19994] [Suggest an Idea]
### Type of discussion.

I'm proposing an idea

### Tell us more.

On Jan 9, a PR was created to update a dependency from 2.8.1 to 2.8.2
On Jan 10, a contributor pushed a fix to the same branch
On Jan 11, the following comment was posted:
```
Edited/Blocked Notification
Renovate will not automatically rebase this PR, because it does not recognize the last commit author and assumes somebody else may have edited the PR.
You can manually request rebase by checking the rebase/retry box above.

⚠️ Warning: custom changes will be lost.
```
On Jan 18, the PR was force-pushed with version 2.8.3, which lost the commit from Jan 10.

This happened because we have a bot that tries to merge the main branch into the PR to ensure tests pass prior to merging. The same GitHub user is being used so that Renovate can rebase and force-push branches that become conflicted. Would it be alright if I create a Pull Request to update the logic in https://github.com/renovatebot/renovate/blob/main/lib/util/git/index.ts#L636 so that it doesn't only look at the last commit and instead, compares the branch to the default branch?

**Comment by viceice** [comment]
no

**Comment by rarkins** [comment]
Ensure gitAuthor is different

**Reply by Keysox** [reply]
Thanks for the response. That would fix the issue for us but would prevent Renovate from rebasing conflicted branches (that have not been modified, with the exception of a merge commit), which we were hoping for. One solution would be to update the `isBranchModified` check with something such as `git log --pretty=format:%ae origin/main..origin/renovate/prettier-2.x --`

**Reply by Keysox** [reply]
@rarkins any additional thoughts on this?

**Reply by rarkins** [reply]
That might be OK. Please propose it as a PR targeting v35 as we will flag it as a breaking change (in theory someone could be relying on the old behavior)

**Comment by cevich** [comment]
I'm seeing this problem as well, though not always because of a dependency update.  It also seems to hapen when the base-branch moves.  Either way, it's really infuriating because sometimes the added commit is really important.  Having renovate clobber it with a force-push risks somebody merging w/o the necessary added commit.  Not to mention that it renders Renovate's message meaningless: "Edited/Blocked Notification...".

An alternate idea to workaround this situation: Support a "renovate-disabled" label, where if present, causes renovate to actually ignore and not touch a PR.
